ValveResourceFormat/Resource/ResourceTypes/Choreo/Data/ChoreoEvent.cs

```diff
@@ -0,0 +1,153 @@
+using ValveResourceFormat.ResourceTypes.Choreo.Enums;
+using ValveResourceFormat.Serialization.KeyValues;
+
+namespace ValveResourceFormat.ResourceTypes.Choreo.Data
+{
+    public class ChoreoEvent
+    {
+        public ChoreoEventType Type { get; private set; }
+        public string Name { get; private set; }
+        public float StartTime { get; private set; }
+        public float EndTime { get; private set; }
+        public string Param1 { get; private set; }
+        public string Param2 { get; private set; }
+        public string Param3 { get; private set; }
+        public ChoreoRamp Ramp { get; private set; }
+        public ChoreoFlags Flags { get; private set; }
+        public float DistanceToTarget { get; private set; }
+        public ChoreoRelativeTag[] RelativeTags { get; private set; }
+        public ChoreoFlexTimingTag[] FlexTimingTags { get; private set; }
+        public ChoreoAbsoluteTag[] AbsoluteTags { get; private set; }
+        public float SequenceDuration { get; private set; }
+        public bool UsingRelativeTag { get; private set; }
+        public ChoreoRelativeTag RelativeTag { get; private set; }
+        public ChoreoEventFlex EventFlex { get; private set; }
+        public byte LoopCount { get; private set; }
+        public ChoreoClosedCaptions ClosedCaptions { get; private set; }
+        public int Id { get; private set; }
+        public int Unk01 { get; private set; }
+
+        //todo: ew
+        public ChoreoEvent(ChoreoEventType type, string name, float startTime, float endTime, string param1, string param2, string param3, ChoreoRamp ramp, ChoreoFlags flags, float distanceToTarget, ChoreoRelativeTag[] relativeTags, ChoreoFlexTimingTag[] flexTimingTags, ChoreoAbsoluteTag[] absoluteTags, float sequenceDuration, bool usingRelativeTag, ChoreoRelativeTag relativeTag, ChoreoEventFlex eventFlex, byte loopCount, ChoreoClosedCaptions closedCaptions, int id, int unk01)
```

Comment:
You can change `private set` to `init` and set them as attributes straight away.